buildpacktoml_test.go

```diff
@@ -0,0 +1,773 @@
+				mkdir(t, filepath.Join(layersDir, "A", "launch.toml"))
+				if _, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config); err == nil {
+					t.Fatal("Expected error")
+				}
+			})
+
+			it("should error when the launch bom has a top level version", func() {
+				mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
+				mkfile(t,
+					"[[bom]]\n"+
+						`name = "some-dep"`+"\n"+
+						`version = "some-version"`+"\n",
+					filepath.Join(appDir, "launch-A-v1.toml"),
+				)
+				_, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+				h.AssertNotNil(t, err)
+				expected := "top level version which is deprecated"
+				h.AssertStringContains(t, err.Error(), expected)
+			})
+
+			it("should error when the build bom has a top level version", func() {
+				mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
+				mkfile(t,
+					"[[bom]]\n"+
+						`name = "some-dep"`+"\n"+
+						`version = "some-version"`+"\n",
+					filepath.Join(appDir, "build-A-v1.toml"),
+				)
+				_, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+				h.AssertNotNil(t, err)
+				expected := "top level version which is deprecated"
+				h.AssertStringContains(t, err.Error(), expected)
+			})
+
+			when("invalid unmet entries", func() {
+				when("missing name", func() {
+					it("should error", func() {
+						mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
+						mkfile(t,
+							"[[unmet]]\n",
+							filepath.Join(appDir, "build-A-v1.toml"),
+						)
+						_, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+						h.AssertNotNil(t, err)
+						expected := "name is required"
+						h.AssertStringContains(t, err.Error(), expected)
+					})
+				})
+
+				when("invalid name", func() {
+					it("should error", func() {
+						mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
+						mkfile(t,
+							"[[unmet]]\n"+
+								`name = "unknown-dep"`+"\n",
+							filepath.Join(appDir, "build-A-v1.toml"),
+						)
+						_, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+						h.AssertNotNil(t, err)
+						expected := "must match a requested dependency"
+						h.AssertStringContains(t, err.Error(), expected)
+					})
+				})
+			})
+		})
+
+		when("buildpack api = 0.2", func() {
+			it.Before(func() {
+				bpTOML.API = "0.2"
+				mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
+			})
+
+			it("should convert metadata version to top level version in the buildpack plan", func() {
+				bpPlan := lifecycle.BuildpackPlan{
+					Entries: []lifecycle.Require{
+						{
+							Name:     "some-dep",
+							Metadata: map[string]interface{}{"version": "v1"},
+						},
+					},
+				}
+
+				_, err := bpTOML.Build(bpPlan, config)
+				if err != nil {
+					t.Fatalf("Unexpected error:\n%s\n", err)
+				}
+
+				testPlan(t,
```

Comment:
`testPlan` is only used in the this test file but is still defined in `builder_test.go`. Maybe we should audit the test functions in `builder_test.go` and move specific ones like this to the file they are used and move generic ones to the `testhelpers` package if appropriate?